internal/commands/app/create_test.go

```diff
@@ -230,222 +230,250 @@ func TestAppCreateHandler(t *testing.T) {
-		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
-		defer teardown()
-		profile.SetRealmBaseURL("http://localhost:8080")
-
-		out, ui := mock.NewUI()
-
-		cmd := &CommandCreate{
-			inputs: createInputs{
-				newAppInputs: newAppInputs{
-					Name:            "test-app",
-					Project:         "123",
-					Location:        realm.LocationVirginia,
-					DeploymentModel: realm.DeploymentModelGlobal,
+			createDisplay: "Realm app based on the Realm app from-app",
+		},
+		{
+			description: "should create a minimal project dry run with cluster set",
+			cluster:     "test-cluster",
+			clients: cli.Clients{
+				Atlas: mock.AtlasClient{
+					ClustersFn: func(groupID string) ([]atlas.Cluster, error) {
+						return []atlas.Cluster{{Name: "test-cluster"}}, nil
+					},
 				},
-				DryRun: true,
 			},
-		}
-
-		assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{}))
-
-		expectedDir := filepath.Join(profile.WorkingDirectory, "test-app")
-		assert.Equal(t, strings.Join([]string{
-			"01:23:45 UTC INFO  A minimal Realm app would be created at " + expectedDir,
-			"01:23:45 UTC DEBUG To create this app run: " + cmd.commandString(true),
-			"",
-		}, "\n"), out.String())
-	})
-
-	t.Run("should create a dry run for the specified from app", func(t *testing.T) {
-		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
-		defer teardown()
-		profile.SetRealmBaseURL("http://localhost:8080")
-
-		out, ui := mock.NewUI()
-
-		testApp := realm.App{
-			ID:          "789",
-			GroupID:     "123",
-			ClientAppID: "from-app-abcde",
-			Name:        "from-app",
-		}
+			createDisplay: "minimal Realm app",
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+			defer teardown()
+			profile.SetRealmBaseURL("http://localhost:8080")
+
+			out, ui := mock.NewUI()
+
+			cmd := &CommandCreate{
+				inputs: createInputs{
+					newAppInputs: newAppInputs{
+						From:            tc.from,
+						Name:            "test-app",
+						Project:         "123",
+						Location:        realm.LocationVirginia,
+						DeploymentModel: realm.DeploymentModelGlobal,
+					},
+					Cluster: tc.cluster,
+					DryRun:  true,
+				},
+			}
 
-		client := mock.RealmClient{}
-		client.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
-			return []realm.App{testApp}, nil
-		}
+			assert.Nil(t, cmd.Handler(profile, ui, tc.clients))
 
-		cmd := &CommandCreate{
-			inputs: createInputs{
-				newAppInputs: newAppInputs{
-					Name:            "test-app",
-					Project:         "123",
-					Location:        realm.LocationVirginia,
-					DeploymentModel: realm.DeploymentModelGlobal,
-					From:            "from-app",
+			expectedDir := filepath.Join(profile.WorkingDirectory, "test-app")
+			display := make([]string, 0, 4)
+			display = append(display, fmt.Sprintf("01:23:45 UTC INFO  A %s would be created at "+expectedDir, tc.createDisplay))
+			if tc.cluster != "" {
```

Comment:
considering we don't have to deal with any fixed width spacing issues here, i'm tempted to suggest using `expectedOutput func(rootDir string) string` in the `tc` struct (or just a separate test func all together) instead of using the `if` check here ... but "tomato tomato" as they say